src/main.rs

```diff
@@ -64,6 +64,30 @@ struct Opt {
     #[argh(positional)]
     /// input file.
     input_file: String,
+
+    #[argh(option)]
+    /// exclude declarations from the generated output.
+    exclude_declaration: Vec<String>,
+}
+
+/// Remove declarations listed in the input filter.
+fn filter_declarations(
+    file: parser::ast::File,
+    exclude_declarations: &[String],
+) -> parser::ast::File {
+    let ast::File { version, file, endianness, comments, declarations } = file;
+    ast::File {
+        version,
+        file,
+        endianness,
+        comments,
+        declarations: declarations
+            .into_iter()
+            .filter(|decl| {
+                decl.id().map(|id| !exclude_declarations.contains(&id.to_owned())).unwrap_or(true)
+            })
+            .collect(),
+    }
```

Comment:
You should be able to use something like `..file` here:

```suggestion
    ast::File {
        declarations: declarations
            .into_iter()
            .filter(|decl| {
                decl.id().map(|id| !exclude_declarations.contains(&id.to_owned())).unwrap_or(true)
            })
            .collect(),
        ..file
    }
```